$NetBSD: patch-CMakeLists.txt,v 1.3 2023/11/17 09:27:51 nia Exp $

- Treat APPLE less specially.
- Use proper dependencies when building libretro core.

--- CMakeLists.txt.orig	2023-10-27 22:19:54.000000000 +0000
+++ CMakeLists.txt
@@ -429,12 +429,6 @@ if(NOT LIBRETRO)
 		endif()
 	endif()
 
-	find_package(ZLIB)
-	if(TARGET ZLIB::ZLIB AND NOT ANDROID AND (NOT WIN32 OR WINDOWS_STORE))
-		set(WITH_SYSTEM_ZLIB ON CACHE BOOL "Use system provided zlib library")
-		target_link_libraries(${PROJECT_NAME} PRIVATE ZLIB::ZLIB)
-	endif()
-
 	find_package(Lua 5.2)
 	if(NOT APPLE AND LUA_FOUND)
 		target_compile_definitions(${PROJECT_NAME} PRIVATE USE_LUA)
@@ -448,6 +442,12 @@ add_subdirectory(core/deps/libchdr EXCLU
 target_link_libraries(${PROJECT_NAME} PRIVATE chdr-static)
 target_include_directories(${PROJECT_NAME} PRIVATE core/deps/libchdr/include)
 
+find_package(ZLIB)
+if(TARGET ZLIB::ZLIB AND NOT ANDROID AND (NOT WIN32 OR WINDOWS_STORE))
+	set(WITH_SYSTEM_ZLIB ON CACHE BOOL "Use system provided zlib library")
+	target_link_libraries(${PROJECT_NAME} PRIVATE ZLIB::ZLIB)
+endif()
+
 if(NOT WITH_SYSTEM_ZLIB)
 	target_include_directories(${PROJECT_NAME} PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/core/deps/libchdr/deps/zlib-1.2.11")
 	target_link_libraries(${PROJECT_NAME} PRIVATE zlib)
@@ -459,7 +459,7 @@ if(NOT WITH_SYSTEM_ZLIB)
 endif()
 
 find_package(PkgConfig)
-if(PKG_CONFIG_FOUND AND NOT ANDROID AND NOT APPLE AND NOT LIBRETRO)
+if(PKG_CONFIG_FOUND AND NOT ANDROID AND NOT LIBRETRO)
 	pkg_check_modules(AO IMPORTED_TARGET ao)
 	if(AO_FOUND)
 		target_compile_definitions(${PROJECT_NAME} PRIVATE USE_LIBAO)
@@ -485,7 +485,9 @@ if(PKG_CONFIG_FOUND AND NOT ANDROID AND 
 		target_compile_definitions(${PROJECT_NAME} PRIVATE USE_PULSEAUDIO)
 		target_link_libraries(${PROJECT_NAME} PRIVATE PkgConfig::LIBPULSE)
 	endif()
+endif()
 
+if(PKG_CONFIG_FOUND)
 	if(USE_HOST_LIBZIP)
 		pkg_check_modules(LIBZIP IMPORTED_TARGET libzip)
 		if(LIBZIP_FOUND)
@@ -510,7 +512,7 @@ if(NINTENDO_SWITCH AND USE_GLES)
 	target_link_libraries(${PROJECT_NAME} PRIVATE PkgConfig::GLESV2)
 endif()
 
-if(UNIX AND NOT APPLE AND NOT ANDROID)
+if(UNIX AND NOT ANDROID)
 	add_definitions(
 		-DFLYCAST_DATADIR="${CMAKE_INSTALL_FULL_DATADIR}/${PROJECT_NAME}/"
 		-DFLYCAST_SYSCONFDIR="${CMAKE_INSTALL_FULL_SYSCONFDIR}/${PROJECT_NAME}/"
@@ -1795,7 +1797,7 @@ else()
 	install(TARGETS ${PROJECT_NAME} DESTINATION "${CMAKE_INSTALL_BINDIR}")
 endif()
 
-if(UNIX AND NOT APPLE AND NOT ANDROID AND NOT LIBRETRO)
+if(UNIX AND NOT ANDROID AND NOT LIBRETRO)
 	install(FILES shell/linux/man/${PROJECT_NAME}.1
 		DESTINATION "${CMAKE_INSTALL_MANDIR}/man1"
 	)
